File: Raspi/client/python3/run_raspi_car.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

"""
@file: run_raspi_car.py
@time: 2018/8/12 0:19
"""

from driver import driver
import time


def run_raspi_car():
    print("==========Client Start==========")
    # if True:
    #    d = driver()
    #    d.open()
    with driver() as d:
        if d is not None:
            d.setStatus(motor=0.0, servo=0.0, dist=0x00, cam_pitch=0, cam_yaw=0, mode="stop")
            t = 10
            b = True
            try:
                print("Test 1/2: voltage mode")
                d.setStatus(mode="voltage")
                while True:
                    st = t * 0.1 - 1
                    sm = st
                    if abs(st) < 0.4:
                        sm = 0.4 if st > 0 else -0.4
                        if st == 0:
                            sm = 0
                    d.setStatus(motor=sm, servo=st)
                    print("Motor: %0.2f, Servo: %0.2f" % (sm, st))
                    if b:
                        t += 1
                    else:
                        t -= 1
                    if t > 20:
                        b = False
                        t -= 2
                    if t < 0:
                        b = True
                        t += 2
                    time.sleep(1)
                    # d.heartBeat()
                    d.getMotor()
                    d.getServo()
                    d.getMode()
                    time.sleep(1)
            except KeyboardInterrupt:
                d.setStatus(motor=0.0, servo=0.0, dist=0x00, mode="stop")
            try:
                print("Test 2/2: camera move")
                d.setStatus(motor=0.0, servo=0, mode="voltage")
                while True:
                    st = t * 0.1 - 1
                    sm = st
                    d.setStatus(cam_pitch=sm, cam_yaw=st)
                    print("Pitch: %0.2f, Yaw: %0.2f" % (sm, st))
                    if b:
                        t += 1
                    else:
                        t -= 1
                    if t > 20:
                        b = False
                        t -= 2
                    if t < 0:
                        b = True
                        t += 2
                    time.sleep(1)
                    # d.heartBeat()
                    d.getCamPitch()
                    d.getCamYaw()
                    time.sleep(1)
            except KeyboardInterrupt:
                d.setStatus(motor=0.0, servo=0.0, dist=0x00, cam_pitch=0, cam_yaw=0, mode="stop")
            d.close()
    print("==========Client Fin==========")
    return 0
               
    
if __name__ == '__main__':
    run_raspi_car()

File: Raspi/client/cpp/run_raspi_car.cpp
#include "driver.h"
#include <unistd.h>
#include <csignal>
#include <cmath>

using namespace std;

bool kp1,kp2;

void sig_handler(int sig){
    if(sig == SIGINT){
        kp1 = !kp1;
        kp2 = !kp2;
    }
}

int main(){
    kp1 = true;
    kp2 = false;
    signal(SIGINT, sig_handler);
    driver* d = new driver();
    d->setMotion(0.0f, 0.0f);
    d->allStop();
    int t = 10;
    bool b = true;
    printf("Test 1/2: voltage mode\n");
    d->launchVoltageMode();
    while(kp1){
        float st = t * 0.1 - 1;
        float sm = st;
        if (abs(st) < 0.4) {
            sm = (st > 0)? 0.4: -0.4;
            if (abs(st) < 0.01){
                sm = 0;
            }
        }
        d->setMotion(sm, st);
        printf("Motor: %0.2f, Servo: %0.2f\n", sm, st);
        if (b){t++;}else{t--;}
        if (t > 20) {b = false; t-=2;}
        if (t < 0)  {b = true; t+=2;}
        sleep(1);
        d->getMotor();
        d->getServo();
        d->getMode();
        sleep(1);
    };
    while(kp2){
        st = t * 0.1 - 1;
        sm = st;
        d->setMotion(sm, st);
        printf("Motor: %0.2f, Servo: %0.2f\n", sm, st);
        if (b){t++;}else{t--;}
        if (t > 20) {b = false; t-=2;}
        if (t < 0)  {b = true; t+=2;}
        sleep(1);
        d->getMotor();
        d->getServo();
        d->getMode();
        sleep(1);
    };
    d->setMotion(0.0, 0.0);
    d->setDistance(0x00);
    d->allStop();
    delete d;
    return 0;
}
